gquant/dataframe_flow/task.py

```diff
@@ -67,6 +68,7 @@ def get_node_obj(self, replace=None, profile=False, tgraph_mixin=False):
                 spec = importlib.util.spec_from_file_location(
                     modulename, modulepath)
                 mod = importlib.util.module_from_spec(spec)
+                sys.modules[spec.name] = mod
```

Comment:
I get an error when running the `test_dask.ipynb` notebook.
```
  File "<my conda python>/lib/python3.7/site-packages/distributed/protocol/pickle.py", line 75, in loads
    return pickle.loads(x)
ModuleNotFoundError: No module named 'custom_port_nodes'
```

To fix this I've modified this part for dask logic to add the path on the workers.
```
            if modulepath is not None:
                spec = importlib.util.spec_from_file_location(node_id,
                                                              modulepath)
                pp = Path(modulepath)
                modulename = pp.stem
                spec = importlib.util.spec_from_file_location(
                    modulename, modulepath)
                mod = importlib.util.module_from_spec(spec)
                sys.modules[spec.name] = mod

                module_dir = str(pp.parent.absolute())

                def append_path(path):
                    if path not in sys.path:
                        sys.path.append(path)

                append_path(module_dir)

                try:
                    import dask.distributed
                    client = dask.distributed.client.default_client()
                    client.run(append_path, module_dir)
                except (ValueError, ImportError):
                    pass

                spec.loader.exec_module(mod)
                NodeClass = getattr(mod, node_type)
            else:
                . . .
```

Do you have this problem of dask not being able to find `custom_port_nodes` module? If I put `custom_port_nodes.py` in the same directory as the notebook then the notebook works fine.

Reply:
Good catch. I took your code and fixed it.